app-libs/stf/src/challenge_code.rs

```diff
@@ -0,0 +1,66 @@
+// Copyright 2020-2022 Litentry Technologies GmbH.
+// This file is part of Litentry.
+//
+// Litentry is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Litentry is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Litentry.  If not, see <https://www.gnu.org/licenses/>.
+
+use rand::Rng;
+
+pub trait ChallengeCodeGenerator {
+	fn generate(&mut self) -> String;
+}
+
+#[derive(Default)]
+pub struct AlphanumericChallengeCode {
+	pub len: usize,
+}
+
+impl ChallengeCodeGenerator for AlphanumericChallengeCode{
+	fn generate(&mut self) -> String {
+		if self.len == 0 {
```

Comment:
Maybe let's remove the if branch here, we'll pass the len 6 as parameter from the constructor. A hardcoded length 6 is a bit weird for a more generic struct..

Reply:
It is just a default value. We can remove it.